### DISCUSSION TITLE
Guidance re accepting a `KeysInterface` and supporting transactions

### DISCUSSION THREAD

**Discussion by maxcountryman** [aembke/fred.rs#227] [Q&A]
I'm writing a crate that ideally accepts `KeysInterface` as opposed to `RedisClient` or `RedisPool` but also needs to leverage transactions. 

Because `TransactionInterface` is not implemented for `RedisPool` this bound, `KeysInterface + TransactionInterface`, doesn't seem to help.

Is there an established pattern for this?

**Comment by aembke** [comment]
Good question. On second thought I'll add `TransactionsInterface` to `RedisPool`. I think the usability benefits there outweigh whatever concerns I had about that before.

**Reply by maxcountryman** [reply]
Thanks, that would be very helpful at least insomuch as my use case is concerned. Do you have an idea of whether this would land in the `8.x` series or a subsequent major release?

**Reply by aembke** [reply]
I'll add this in 8.0.6 sometime over the weekend.

**Reply by maxcountryman** [reply]
~~Hi @aembke, I was hoping to check if this is still planned for the near future?~~

Edit, I should have checked [the PRs](https://github.com/aembke/fred.rs/pull/230) first. 😄

**Reply by aembke** [reply]
Ha no worries. I just pushed it now to crates.io as well.

For what it's worth I think I'll also have to do a 9.0.0 soon to cut over to MSRV 1.75, AFIT, and rustls 0.23. Just in case that matters for your versioning purposes.
